packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp

```diff
@@ -0,0 +1,269 @@
+@added(Versions.v5)
+union CreatedByType {
+  string,
+  User: "User",
+  Application: "Application",
+  ManagedIdentity: "ManagedIdentity",
+  Key: "Key",
+}
+
+/** Allow, disallow, or let network security perimeter configuration control public network access to the protected resource. Value is optional but if passed in, it must be 'Enabled', 'Disabled' or 'SecuredByPerimeter'. */
+@added(Versions.v5)
+union PublicNetworkAccess {
+  string,
+
+  /** Allows public network access to the resource */
+  Enabled: "Enabled",
+
+  /** Disallows public network access to the resource */
+  Disabled: "Disabled",
+
+  /** The network security perimeter configuration rules allow or disallow public network access to the resource. Requires an associated network security perimeter. */
+  SecuredByPerimeter: "SecuredByPerimeter",
+}
+
+/** Access rule in a network security perimeter configuration profile */
+@added(Versions.v5)
+model AccessRule {
+  /** Name of the access rule */
+  name?: string;
+
+  properties?: AccessRuleProperties;
+}
+
+/** Properties of Access Rule */
+@added(Versions.v5)
+model AccessRuleProperties {
+  direction?: AccessRuleDirection;
+
+  /** Address prefixes in the CIDR format for inbound rules */
+  addressPrefixes?: string[];
+
+  /** Subscriptions for inbound rules */
+  subscriptions?: {
+    /** The fully qualified Azure resource ID of the subscription e.g. ('/subscriptions/00000000-0000-0000-0000-000000000000') */
+    id?: Azure.Core.armResourceIdentifier;
+  }[];
+
+  /** Network security perimeters for inbound rules */
+  networkSecurityPerimeters?: NetworkSecurityPerimeter[];
+
+  /** Fully qualified domain names (FQDN) for outbound rules */
+  fullyQualifiedDomainNames?: string[];
+
+  /** Email addresses for outbound rules */
+  emailAddresses?: string[];
+
+  /** Phone numbers for outbound rules */
+  phoneNumbers?: string[];
+}
+
+/** Information about a network security perimeter (NSP) */
+@added(Versions.v5)
+model NetworkSecurityPerimeter {
+  /** Fully qualified Azure resource ID of the NSP resource */
+  id?: Azure.Core.armResourceIdentifier<[
+    {
+      type: "Microsoft.Network/networkSecurityPerimeters";
+    }
+  ]>;
+
+  /** Universal unique ID (UUID) of the network security perimeter */
+  perimeterGuid?: uuid;
+
+  /** Location of the network security perimeter */
+  @visibility("read", "create")
+  location?: string;
+}
+
+/** Network security perimeter (NSP) configuration resource */
+@added(Versions.v5)
+model NetworkSecurityPerimeterConfiguration extends ProxyResource {
+  properties?: NetworkSecurityPerimeterConfigurationProperties;
+}
+
+/** Network security configuration properties. */
+@added(Versions.v5)
+model NetworkSecurityPerimeterConfigurationProperties {
+  @visibility("read")
+  provisioningState?: NetworkSecurityPerimeterConfigurationProvisioningState;
+
+  /** List of provisioning issues, if any */
+  @visibility("read")
+  @OpenAPI.extension("x-ms-identifiers", [])
```

Comment:
```suggestion
  @OpenAPI.extension("x-ms-identifiers", ["name"])
```

Reply:
I think this one is empty: https://github.com/Azure/azure-rest-api-specs/blob/34f5146bc945549d087d38a8a593c0a5f475ad7f/specification/common-types/resource-management/v5/networksecurityperimeter.json#L107

Reply:
Yeah, it probably should not be, but we should match